upstairs/src/guest.rs

```diff
@@ -127,20 +90,45 @@ impl GtoS {
          * given up because an IO took too long, or other possible
          * guest side reasons.
          */
-        if let Some(res) = self.res {
-            match result {
-                Ok(_) => match guest_buffer {
-                    Some(guest_buffer) => res.send_ok_with_buffer(guest_buffer),
-                    None => res.send_ok(),
-                },
-
-                Err(e) => match guest_buffer {
-                    Some(guest_buffer) => {
-                        res.send_err_with_buffer(guest_buffer, e)
+        match self.res {
+            Some(GuestBlockRes::Read(mut buffer, res)) => {
+                if let Some(downstairs_responses) = downstairs_responses {
+                    let mut offset = 0;
+
+                    // XXX don't do if result.is_err()?
+                    for response in &downstairs_responses {
+                        // Copy over into guest memory.
+                        {
+                            let _ignored =
+                                span!(Level::TRACE, "copy to guest buffer")
+                                    .entered();
+
+                            buffer.write_read_response(offset, response);
+                            offset += response.data.len();
+                        }
                     }
-                    None => res.send_err(e),
-                },
+                } else {
+                    /*
+                     * Should this panic?  If the caller is requesting a
+                     * transfer, the guest_buffer should exist. If it does not
+                     * exist, then either there is a real problem, or the
+                     * operation was a write or flush and why are we requesting
+                     * a transfer for those.
+                     *
+                     * However, dropping a Guest before receiving a downstairs
+                     * response will trigger this, so eat it for now.
+                     */
```

Comment:
style nit: convert this to `//`-style comment, since it's being altered/moved?

Reply:
Sure, done in 515ae69e5d0251bc5062e9a9411f7739a7f85a19